### PR TITLE
k8 deployment retry for gcloud

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@scopatz ready for review!
Or should we do this with https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-restart-em-

File: news/kube_retry.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+**Added:**
+
+* `rever.activities.gcloud.DeploytoGcloud` now has an undo
```
Comment (scopatz) [inline news/kube_retry.rst]: Double backticks needed in this file

File: rever/activities/gcloud.xsh [hunk 2]
```diff
@@ -69,8 +69,31 @@ class DeploytoGCloud(Activity):
...
+        # make certain kubectl rolls out, otherwise raise exception
+        for i in range(3):
+            # set new image
+            ![kubectl set image deployment/$GCLOUD_CONTAINER_NAME $GCLOUD_CONTAINER_NAME=$GCLOUD_DOCKER_HOST/$GCLOUD_DOCKER_ORG/$GCLOUD_DOCKER_REPO:$VERSION]
+            if ![kubectl rollout status deployment/$GCLOUD_CONTAINER_NAME]:
+                break
+        else:
+            raise RuntimeException('kubectl failed to rollout the new image')
+
+    def undo(self,
```
Comment (scopatz) [inline rever/activities/gcloud.xsh]: Awesome! Thanks for adding the undo

**Comment by scopatz** [comment]
I think this is good once the rst is fixed

File: news/kube_retry.rst (hunk 1 above)
Comment (CJ-Wright) [inline news/kube_retry.rst]:
```suggestion
* ``rever.activities.gcloud.DeploytoGcloud`` now has an undo
```

**Comment by CJ-Wright** [comment]
@scopatz ready for another round.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**